### PR TITLE
Add CoreDNS configmap to federation test

### LINKED ISSUE #67600: Failing Test: DNS configMap federations should be able to change federation configuration
# Failing Jobs
[sig-release-master-blocking#gci-gce-serial](https://k8s-testgrid.appspot.com/sig-release-master-blocking#gci-gce-serial)
# Failing Tests
[[sig-network] DNS configMap federations should be able to change federation configuration [Slow][Serial]](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/3980)
# Triage Results
[[sig-network] DNS configMap federations should be able to change federation configuration [Slow][Serial]](https://storage.googleapis.com/k8s-gubernator/triage/index.html?job=ci-kubernetes-e2e-gci-gce-serial&test=DNS%20configMap%20federations%20should%20be%20able%20to%20change%20federation%20configuration)

/kind failing-test
/priority critical-urgent
/sig network
/milestone v1.12
/cc @tpepper @AishSundar

### PR DESCRIPTION
**Pull request by rajansandeep**
**What this PR does / why we need it**:
Fix the federation test by adding the correct CoreDNS configmap.
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #67600 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/dns_configmap.go [hunk 1]
```diff
@@ -57,39 +57,96 @@ func (t *dnsFederationsConfigMapTest) run() {
...
+           fallthrough in-addr.arpa ip6.arpa
+        }
+         federation {
+           ghi xyz
+         }
+        
+    }`}
+
+		By("default -> valid1")
+		t.setConfigMap(&v1.ConfigMap{Data: valid1}, nil, true)
```
[the remark this answers is not in the document]
Comment (rajansandeep) [reply test/e2e/network/dns_configmap.go]: fixed it.

**Comment by rajansandeep** [comment]
@MrHohn This is ready for final review. PTAL

**Review by MrHohn — APPROVED** [review]
Thanks!
/lgtm

**Pull request merged**